src/Arcus.WebApi.Unit/Correlation/CorrelationController.cs

```diff
@@ -14,23 +15,29 @@ public class CorrelationController : ControllerBase
     {
         public const string Route = "correlation";
 
-        private readonly HttpCorrelationInfo _correlationInfo;
+        private readonly ICorrelationInfoAccessor _correlationInfoAccessor;
         private readonly DiagnosticContext _diagnosticContext;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="CorrelationController"/> class.
         /// </summary>
-        public CorrelationController(HttpCorrelationInfo correlationInfo, DiagnosticContext diagnosticContext)
+        public CorrelationController(ICorrelationInfoAccessor correlationInfoAccessor, DiagnosticContext diagnosticContext)
         {
-            _correlationInfo = correlationInfo;
+            _correlationInfoAccessor = correlationInfoAccessor;
             _diagnosticContext = diagnosticContext;
         }
 
         [HttpGet]
         [Route(Route)]
         public IActionResult Get()
         {
-            string json = JsonConvert.SerializeObject(new { _correlationInfo.TransactionId, _correlationInfo.OperationId });
+            var correlationInfoModel = new
+            {
+                _correlationInfoAccessor.CorrelationInfo.TransactionId, 
+                _correlationInfoAccessor.CorrelationInfo.OperationId
+            };
+            string json = JsonConvert.SerializeObject(correlationInfoModel);
```

Comment:
Just serialize this and remove the new object?
```suggestion
            string json = JsonConvert.SerializeObject(_correlationInfoAccessor.CorrelationInfo);
```